packages/main-api/server/utils.js

```diff
@@ -0,0 +1,9 @@
+// async/await error catcher
+const catchAsyncErrors = (fn) => (req, res, next) => {
+  const routePromise = fn(req, res, next);
+  if (routePromise.catch) {
```

Comment:
catch is always present on a promise, and any async function returns a promise. This check is not needed. If this is to make sure devs don't make mistakes then a typing system does it way better.

Reply:
I wouldn't worry about this for now, this is just a placeholder. I'm going to do a live code review with him @mmcken3, I just moved his stuff into the repo.